### PR TITLE
Python 3 and PyPy support.

### PR DESCRIPTION
**Pull request by jamadden**
Also modernize project:
- Remove zope.app.zcmlfiles and zope.app.testing deps, among others.
- tox.ini
- .travis.yml uses pip, zope.testrunner and supports caching and
coverage
- Rename .txt -> .rst

### CODE REVIEW

File: src/zope/app/exception/browser/systemerror.rst [hunk 1]
```diff
@@ -12,10 +12,10 @@ terse message and that sets the response status.
 There is a simple view registered in ``ftesting.zcml`` which raises
 ``Exception()``:
 
-  >>> print http(r"""
+  >>> print(http(r"""
   ... GET /error.html HTTP/1.1
-  ... """)
-  HTTP/1.1 500 Internal Server Error
+  ... """))
+  HTTP/1.0 500 Internal Server Error
```
Comment (icemac) [inline src/zope/app/exception/browser/systemerror.rst:18]: Why is this now `1.0`? Any clue?

File: src/zope/app/exception/browser/tests/test_unauthorized.py [hunk 2]
```diff
@@ -43,23 +43,23 @@ def unauthorized(self, principal_id, request):
...
         return 'You are not authorized'
 
-class Test(PlacelessSetup, TestCase):
+class Test(PlacelessSetup, unittest.TestCase):
 
     def setUp(self):
         super(Test, self).setUp()
         self.auth = DummyAuthUtility()
-        ztapi.provideUtility(IAuthentication, self.auth)
+        component.provideUtility(self.auth, IAuthentication)
```
Comment (icemac) [inline src/zope/app/exception/browser/tests/test_unauthorized.py:62]: This should be torn down, too.

**Review by icemac — APPROVED** [review]
LGTM.

File: src/zope/app/exception/browser/systemerror.rst (hunk 1 above)
Comment (jamadden) [reply src/zope/app/exception/browser/systemerror.rst:18]: One of the differences between `zope.app.testing` and `zope.app.wsgi.testlayer` is this. I've seen that on each of the projects I've updated.

File: src/zope/app/exception/browser/tests/test_unauthorized.py (hunk 2 above)
Comment (jamadden) [reply src/zope/app/exception/browser/tests/test_unauthorized.py:62]: I believe it is. The test class extends `zope.component.testing.PlacelessSetup` and makes sure to call its `tearDown` method, which should clean this all up.

**Comment by jamadden** [comment]
Thanks for the review! A PyPI release would be helpful when convenient.

**Comment by icemac** [comment]
@jamadden You now have the necessary rights on PyPI.

**Comment by jamadden** [comment]
Thank you! 4.0.0 has been released to PyPI.

**Pull request merged**